packages/dds/tree/src/test/TypedSchema.spec.ts

```diff
@@ -0,0 +1,44 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+/* eslint-disable import/no-internal-modules */
+
+// TODO: what is the pattern for testing packages which have folders inside src?
+import { FieldKind, ValueSchema } from "../schema/Schema";
```

Comment:
usually an index per folder, and then the top-level index re-exports those indicies. not sure if eslint will be happy about that though

Reply:
I like to write a unit test file for most source files and import that source file directly in the unit test file: I find this the most readable/maintainable. If we put this test file in the top level test directory, its has to do this import which es-lint objects to.

I'm aware of the index reexporting pattern, and I like it for non-test APIs. But for unit testing, that would involve exporting any test only APIs through 2 levels of index.ts files exposing all the test APIs in both the API for the inner folder and the package public API right? This seems bad. It also makes it less clear which things are supposed to be tested by the test file (since all the package imports are grouped together under `..`). If we used API-extractor's scope tags, this would be less of an issue, but still not ideal.

Another approach would be to move the test closer to the source (src/schema/test in this case), but that approach doesn't work very clearly with our typescript project setup for testing (ex: vs code intelisense will infer the wrong project unless we add a tsconfig for every test folder)

At this stage of development `schema` doesn't actually have any public APIs, just unit tests, so its index file is empty/missing, and I'm trying to take a future proof approach to testing that doesn't rely on adding things to the public APIs.

Reply:
I fixed the internal import from the file not being tested here, made the list disable next line only and documented the testing pattern being used here.